src/App.js

```diff
@@ -1,12 +1,117 @@
-import React from 'react';
+import React, { Component } from 'react';
+import PostList from './PostList/PostList';
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Dynamic list of posts</h1>
-    </div>
-  );
+class App extends Component {
+  state ={
+    isLoading: false,
+    isLoaded: false,
+    isFiltered: false,
+    inputValue: '',
+  }
+
+  getData = () => {
+    this.setState({
+      isLoading: true,
+    });
+
+    Promise.all([
+      fetch('https://jsonplaceholder.typicode.com/posts'),
+      fetch('https://jsonplaceholder.typicode.com/users'),
+      fetch('https://jsonplaceholder.typicode.com/comments'),
+    ])
+      .then(([posts, users, comments]) => Promise.all([
+        posts.json(),
+        users.json(),
+        comments.json(),
+      ]))
+      .then(([posts, users, comments]) => {
+        const usersMap = users.reduce((acc, user) => ({
+          ...acc,
+          [user.id]: user,
+        }), {});
+
+        const preparedPosts = posts
+          .map(post => ({
+            ...post,
+            comments: comments.filter(comment => comment.postId === post.id),
+            user: usersMap[post.userId],
+          }));
+
+        this.setState({
+          posts: preparedPosts,
+          filteredPosts: preparedPosts,
+          isLoaded: true,
+        });
+      });
+  }
+
+  filter = (event) => {
```

Comment:
give more specific name

Reply:
imagine you have many functions with different purposes on a page, you need to name them the way so that you can differentiate them

and even if you have only one filtering function, it's a good practice to give a name that would describe a function purpose the best